mne/annotations.py

```diff
@@ -700,16 +700,60 @@ def _ensure_annotation_object(obj):
         raise ValueError('Annotations must be an instance of '
                          'mne.Annotations. Got %s.' % obj)
 
+def _select_annotations_based_on_description(descriptions, event_id=None,
+                                             regexp=None):
+    """Gets a collection of descriptions and returns index of selected.
+    """
+    # Filter out the annotations that do not match regexp
+    regexp_comp = re.compile('.*' if regexp is None else regexp)
```

Comment:
I don't get what you mean. If I change the function's signature from `regexp=None` to `regexp='.*'` then I'll have to make the check outside and only add the regex parameter if it is not none; and we rather prefer the inner function doing that, wouldn't we?

Reply:
With @agramfort's suggestion you never need to check `if regexp is None` anymore. What check will you still need to do?

Reply:
To avoid this situation:
```
def events_from_annotations(raw, event_id=None, regexp=None, use_rounding=True,
                            verbose=None):
    ...
    _select_annotations_based_on_description(descriptions, regexp=regexp)
    ...

def _select_annotations_based_on_description(descriptions, event_id=None,
                                             regexp='.*'):
     assert regexp is None
```

Otherwise, I've to deprecate the `regexp=None`

Reply:
Instead of `assert regexp is None` inside `_select_annotataions...`, did you mean the opposite, `assert regexp is not None`?

I agree with @agramfort that `regexp='.*'` is a nicer default than `None`. And it simplifies the inner function to know that `regexp` is a string. So I would do:
```
def events_from_annotations(..., regexp='.*', ...)
    """...
    regexp : str
        ...
    """
    # regexp None support for 0.17 backward compat, not exposed to users
    regexp = '.*' if regexp is None else regexp
    _validate_type(regexp, str, 'regexp')

...
def _select_annotations_based_on_description(descriptions, event_id, regexp):
     assert isinstance(regexp, str)
```

Reply:
The assert was just to illustrate. If we don't have to deprecate the old behavior with a deprecation cycle and everything. Then I agree.

Reply:
We shouldn't need to since `'.*'` will match everything just like `None` would have.

The only difference is whether an error is raised if no events are found. I'm not sure we should have ever differentiated based on whether `regexp` was passed by the user or not. The most consistent thing to do would be to have an `on_missing` argument like we do for `Epochs`. I think it's okay to change this behavior without deprecation since it's going to be a corner case.